### PR TITLE
Feature: Implemented the Custom Error Screen

### LINKED ISSUE #133: Feature: Implementing Custom Error Screen
### Description

Currently, if any error/exception is thrown, a red screen appears in the UI. It's not good at all in terms of the user experience. So I think we should implement a custom error screen that shows the user what can be the issue.

_I would like to work on this, so please assign it to me._

### Screenshots

_No Response_

### PR DESCRIPTION
**Pull request by superiorsd10**
### Description
Previously, whenever any error/exception was thrown, a red/grey screen appeared in the UI. It was not good in terms of UX. So I've implemented the custom error screen that shows the summary of error/exception in the debug mode, and a message in the release mode. 

### Proposed Changes
- Added the `CustomErrorScreen` in the `utils`.
- Added the `builder` function to the `GetMaterialApp`.

## Checklist

- [ ] Tests have been added or updated to cover the changes
- [ ] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines
- [ ] All tests are passing

### CODE REVIEW

**Comment by MarkisDev** [comment]
Could we also have a title/heading signifying its an error? Seems a bit weird without it.

File: lib/app/utils/custom_error_screen.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          ? kLightPrimaryBackgroundColor
+          : kprimaryBackgroundColor,
+      body: SafeArea(
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: EdgeInsets.fromLTRB(16, height * 0.2, 16, 16),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                SvgPicture.asset(
```
Comment (MarkisDev) [inline lib/app/utils/custom_error_screen.dart:38]: A title/heading signifying an error would be great!

**Review by MarkisDev — CHANGES_REQUESTED** [review]

File: lib/app/utils/custom_error_screen.dart (hunk 1 above)
Comment (superiorsd10) [reply lib/app/utils/custom_error_screen.dart:38]: Okay, do you mean an "Error Occured" title at the top?

Comment (MarkisDev) [reply lib/app/utils/custom_error_screen.dart:38]: Yeah

**Comment by superiorsd10** [comment]
Hey @MarkisDev 👋
I've added the title `Error Occurred` at the top of the screen to depict that an error has occurred.

**Review by MarkisDev — APPROVED** [review]

**Pull request merged**